src/queueing_honey_badger/mod.rs

```diff
@@ -127,7 +143,10 @@ where
             queue: self.queue,
             rng: Box::new(rng),
         };
-        let step = qhb.propose()?;
+        let mut step = qhb.propose()?;
+        if let Some(dhb_step) = self.step {
+            step.extend_with(dhb_step, Message::from);
```

Comment:
The types are subtly different: `DaStep<SenderQueue<QHB>>` and `DaStep<QHB>`. So, conversion is still required. (And I have to remind myself not to use `Step::extend_with` because it discards the output...)

Reply:
Right, we should probably make `extend_with` `#[must_use]`.